language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandClientConfig.java

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava.grpc;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Optional.empty;
+
+import com.daml.ledger.javaapi.data.codegen.HasCommands;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * TODO: javadoc this
+ * This class represents a set of parameters required for {@link com.daml.ledger.rxjava.CommandClient}
+ *
+ * Usage:
+ * <pre>
+ *   var params = CommandClientConfig.create("workflowId", "applicationId", "commandId")
+ *                                   .params.withAccessToken("token")
+ *
+ *
+ * <pre/>
+ */
+public class CommandClientConfig {
+  private String workflowId;
+  private String applicationId;
+  private String commandId;
+  private List<@NonNull String> actAs;
+  private List<@NonNull String> readAs;
+  private Optional<Instant> minLedgerTimeAbs;
+  private Optional<Duration> minLedgerTimeRel;
+  private Optional<Duration> deduplicationTime;
+  private List<@NonNull ? extends HasCommands> commands;
+  private Optional<String> accessToken;
+
+  private CommandClientConfig(
+      String workflowId,
+      String applicationId,
+      String commandId,
+      List<@NonNull String> actAs,
+      List<@NonNull String> readAs,
+      Optional<Instant> minLedgerTimeAbs,
+      Optional<Duration> minLedgerTimeRel,
+      Optional<Duration> deduplicationTime,
+      List<@NonNull ? extends HasCommands> commands,
+      Optional<String> accessToken) {
+    this.workflowId = workflowId;
+    this.applicationId = applicationId;
+    this.commandId = commandId;
+    this.actAs = actAs;
+    this.readAs = readAs;
+    this.minLedgerTimeAbs = minLedgerTimeAbs;
+    this.minLedgerTimeRel = minLedgerTimeRel;
+    this.deduplicationTime = deduplicationTime;
+    this.commands = commands;
+    this.accessToken = accessToken;
+  }
+
+  /**
+   * TODO: java doc this
+   *
+   * @param workflowId
+   * @param applicationId
+   * @param commandId
+   * @return
+   */
+  public static CommandClientConfig create(
```

Comment:
This method should create a valid command, i.e. all required commands need to be filled in before you get to the fluent API:
* The commands must be non-empty, so you should pass at least one command and leave room for more (e.g. asking for at least one command and leaving further ones as variadic parameters)
* On the contrary, the workflow ID is optional, so you don't need to specify it as part of the `create` method.

You can use [this](https://docs.daml.com/app-dev/grpc/proto-docs.html#com-daml-ledger-api-v1-commands) as a reference for optional and required fields. I realize some of them are not exactly trivial, in which case I would recommend you reach out to the Canton Participant team to get clarity.